### DISCUSSION TITLE
Mix Materials in one Print without AMS

### DISCUSSION THREAD

**Discussion by tommy536** [OrcaSlicer/OrcaSlicer#977] [General]
With Orcaslicer, is it possible to change the Filament in one Print for ex. ABS to PLA?

I do not have a AMS to do this. (Voron 2.4)

**Comment by gasprints** [comment]
This took me a while to figure out at first, but you need to have multiple filaments added before you can add a "swap filament" command.  On top of that, you have to slice the file every time before you can accurately see where you want to add another swap by right clicking on the slider in the layer you want the swap to happen.
<img width="420" alt="Screenshot 2023-05-11 at 10 39 06 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/128255447/bd03922e-78e1-431b-858b-e042072e6238">
<img width="293" alt="Screenshot 2023-05-11 at 10 39 38 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/128255447/44ba3a38-3f71-45b9-8f6a-c91da330a999">

**Reply by Mario-vminion** [reply]
and when you add change filament ...and where will the print head goto when you have a filament change ...

**Reply by hliebscher** [reply]
@Mario-vminion  you have to add in printer config a gcode in section filament change 
the Move  Position that you want G0 X... Y .... an then Pause Gcode for your printer
change the filament manual and Resume the print....

**Reply by Mario-vminion** [reply]
And when i use the pause instead of the filament change like in the picture above

**Comment by enewbold** [comment]
I would simply like to change filaments in order to change to a different color.  I do not see any options to do this.  What would a sample of gcode look like to insert at the target layer?  I don't have a "chage filament" option appear.  See attached picture.  Thank you.
![Orca-1](https://github.com/SoftFever/OrcaSlicer/assets/5244831/2e8411d8-829b-4791-b0d2-2c8628372121)

**Reply by LJKraus** [reply]
First, you need two (or more) filaments.
![image](https://github.com/SoftFever/OrcaSlicer/assets/139707784/82802389-5a5f-452d-95b1-9e292d180db6)

Slice, then lower the layer slider to the desired level, and right-click the slider.
![image](https://github.com/SoftFever/OrcaSlicer/assets/139707784/3c99037c-935d-43f8-9ecf-c600957d430f)

Re-slice.


I've been simply inserting pause at the correct layer and then changing filament before resuming on my X1C without AMS, but it's nice to see the change in the slicer.

**Comment by atl285** [comment]
The problem with using "Add Pause" is, that you can change the filament / color only for the whole layer. If you want more, then the right way is, to define the change filament code for the printer (the gcode for "pause print" should be okay). Then add different colors / filaments as described above and now you can use the color tool to paint the parts of the model using the specific color / filament. So you can also change the color within a layer.
